priv/templates/thesis.install/add_a_unique_index_on_slug.exs

```diff
@@ -0,0 +1,20 @@
+defmodule <%= base %>.Repo.Migrations.AddAUniqueIndexOnSlug do
+  @moduledoc false
+  use Ecto.Migration
+
+  def up do
+    # Rreate old non-unique index
+    drop index(:thesis_pages, [:slug])
+
+    # Create a Unique Index
+    create unique_index(:thesis_pages, [:slug])
```

Comment:
This will fail in code bases that already include duplicate slugs. We should probably fail with a friendly error here. You can [detect duplicates](https://stackoverflow.com/a/28157109/204044) or just catch the error and fail gracefully (recommended).